File: cloud/acp-admin-cloud-dependencies/src/main/java/pers/acp/admin/common/base/BaseRepository.java
package pers.acp.admin.common.base;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;

import java.io.Serializable;

/**
 * @author zhang by 15/01/2019
 * @since JDK 11
 */
public interface BaseRepository<T, ID extends Serializable> extends JpaSpecificationExecutor<T>, JpaRepository<T, ID> {
}

File: cloud/acp-admin-cloud-dependencies/src/main/kotlin/pers/acp/admin/common/base/BaseRepository.kt
package pers.acp.admin.common.base

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.data.repository.NoRepositoryBean

/**
 * @author zhang by 15/01/2019
 * @since JDK 11
 */
@NoRepositoryBean
interface BaseRepository<T, ID> : JpaSpecificationExecutor<T>, JpaRepository<T, ID>
